kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaTopicManager.java

```diff
@@ -110,33 +109,32 @@ public void updateCtx() {
     // topicName is in pulsar format. e.g. persistent://public/default/topic-partition-0
     // future will complete with null when topic not owned by this broker, or meet error.
     public CompletableFuture<KafkaTopicConsumerManager> getTopicConsumerManager(String topicName) {
-        return consumerTopicManagers.computeIfAbsent(
-            topicName,
-            t -> {
-                CompletableFuture<PersistentTopic> topic = getTopic(t);
-                checkState(topic != null);
-
-                return topic.thenApply(t2 -> {
-                    if (log.isDebugEnabled()) {
-                        log.debug("[{}] Call getTopicConsumerManager for {}, and create TCM for {}.",
-                            requestHandler.ctx.channel(), topicName, t2);
-                    }
-
-                    if (t2 == null) {
-                        return null;
-                    }
-                    // return consumer manager
-                    return new KafkaTopicConsumerManager(requestHandler, t2);
-                });
+        CompletableFuture<KafkaTopicConsumerManager> consumerManagerFuture = new CompletableFuture<>();
+        if (consumerTopicManagers.containsKey(topicName)) {
+            return consumerTopicManagers.get(topicName);
+        } else {
+            CompletableFuture<PersistentTopic> topic = getTopic(topicName);
+            if (topic != null) {
+                consumerManagerFuture.complete(new KafkaTopicConsumerManager(requestHandler, topic.join()));
+                consumerTopicManagers.put(topicName, consumerManagerFuture);
+            } else {
```

Comment:
I think `getTopic` never returns null, right? If any exception occurs, it returns a future that is completed with null.

Reply:
yes, it returns a future that is completed with null, it would never return null. And I will reconsider `getTopicConsumerManager`.